utils_script/find_detection_score_threshold.py

```diff
@@ -0,0 +1,49 @@
+"""
+Find the best confidence score threshold for each class in object detection tasks
+Use this script when you have a trained model and want to analyze / optimize its performance
+The thresholds can be used later when performing NMS
+Usage is similar to src/super_gradients/examples/evaluate_from_recipe_example/evaluate_from_recipe.py
+
+Notes:
+    This script does NOT run TRAINING, so make sure in the recipe that you load a PRETRAINED MODEL
+    either from one of your checkpoint or from a pretrained model.
+
+General use: python find_detection_score_threshold.py --config-name="DESIRED_RECIPE" architecture="DESIRED_ARCH"
+            checkpoint_params.pretrained_weights="DESIRED_DATASET"
+
+Example: python find_detection_score_threshold.py --config-name=coco2017_yolox architecture=yolox_n
+            checkpoint_params.pretrained_weights=coco
+"""
+
+import hydra
+import pkg_resources
+
+from omegaconf import DictConfig
+
+from super_gradients.training.datasets.datasets_conf import COCO_DETECTION_CLASSES_LIST
+from super_gradients.common.environment.cfg_utils import add_params_to_cfg
+from super_gradients import Trainer, init_trainer
+
+
+@hydra.main(config_path=pkg_resources.resource_filename("super_gradients.recipes", ""), version_base="1.2")
+def main(cfg: DictConfig) -> None:
+    add_params_to_cfg(cfg.training_hyperparams.valid_metrics_list[0].DetectionMetrics, params=["calc_best_score_thresholds=True"])
+    _, valid_metrics_dict = Trainer.evaluate_from_recipe(cfg)
+
+    class_names = COCO_DETECTION_CLASSES_LIST  # change this line to use a different dataset
```

Comment:
Great work. I really love the way to find thresholds using command-line args approach: ` python find_detection_score_threshold.py --config-name=coco2017_yolox architecture=yolox_n
            checkpoint_params.pretrained_weights=coco`
However it seems that for other dataset one have to change the hard-coded `class_names` constant. Is it possible to improve this part and move class names to dataset_params maybe?

Also, consider moving this script to `super_gradients.scripts` to enable using this script as follows:
`python -m super_gradients.find_detection_score_threshold --config-name=coco2017_yolox architecture=yolox_n checkpoint_params.pretrained_weights=coco`